### PR TITLE
Update Springfox to 2.4.0 and fix #3018

### LINKED ISSUE #3018: [Gateway] /swagger-resources conflicts with Springfox
Just for clarity, I open this issue to track a problem that was reported as part of the discussion in #2990.

To clarify: the gateway overrides a URL /swagger-resources provided by Springfox but sometimes it does not work. after discussion with Springfox team we have[ a clean solution](springfox/springfox#1191) that requires version 2.4.0 of Springfox which should be released within days.

Once 2.4.0  is released,we can merge my PR #3019

### PR DESCRIPTION
**Pull request by gmarziou**
Implement  SwaggerResourcesProvider interface to provide /swagger-resources with routed micro services

**Can be merged only when Springfox 2.4.0 is released.**

[ci skip]

Fix #3018

### CODE REVIEW

File: generators/server/templates/src/main/java/package/config/apidoc/_GatewaySwaggerResourcesProvider.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package com.mycompany.myapp.config.apidoc;
```
Comment (jdubois) [inline generators/server/templates/src/main/java/package/config/apidoc/_GatewaySwaggerResourcesProvider.java]: you need to put the package name variable here

Comment (gmarziou) [inline generators/server/templates/src/main/java/package/config/apidoc/_GatewaySwaggerResourcesProvider.java]: Thanks, I will do it tonight

**Comment by gmarziou** [comment]
Springfox 2.4.0 has been released !

https://github.com/springfox/springfox/issues/1201

**Comment by deepu105** [comment]
npm tests failing

**Comment by gmarziou** [comment]
I'll have a look tonight

**Comment by deepu105** [comment]
I guess you created the branch in the main repo

**Comment by gmarziou** [comment]
Sorry, probably after the rebase to resolve conflict.

**Pull request merged**